internal/internal_workflow_test.go

```diff
@@ -1406,6 +1406,19 @@ type updateCallback struct {
 	accept   func()
 	reject   func(error)
 	complete func(interface{}, error)
+	env      *TestWorkflowEnvironment
```

Comment:
Hmm so this class is not for users, this whole file is for testing since it is post-fixed with `*_test.go`. The logic can't be in the updateCallback as we take an impl. from user. The logic needs to be in `(env *testWorkflowEnvironmentImpl) updateWorkflow`. We could potentially add a wrapper around the user interface. Does that make sense?

Reply:
Is `user` in this context referring to the user of the test suite?

Reply:
> this whole file is for testing since it is post-fixed with *_test.go. The logic can't be in the updateCallback as we take an impl. from user
And would it be accurate to rephrase this as "we need to implement this within the test suite, not the individual test logic. `updateCallback` is a test specific implementation"?

Reply:
> We could potentially add a wrapper around the user interface. 
What interface are you referring to here?

Reply:
>Is user in this context referring to the user of the test suite?

Yeah 

>And would it be accurate to rephrase this as "we need to implement this within the test suite, not the individual test logic. updateCallback is a test specific implementation"?

yeah, exactly

> What interface are you referring to here?

`UpdateCallbacks`